python-avd/pyavd/_eos_designs/structured_config/underlay/router_isis.py

```diff
@@ -76,11 +76,11 @@ def router_isis(self: AvdStructuredConfigUnderlay) -> dict | None:
 
     @cached_property
     def _isis_net(self: AvdStructuredConfigUnderlay) -> str | None:
-        if get(self._hostvars, "isis_system_id_format", default="node_id") == "node_id":
+        if get(self._hostvars, "isis_system_id_format") == "node_id":
             isis_system_id_prefix = get(self.shared_utils.switch_data_combined, "isis_system_id_prefix")
-            if isis_system_id_prefix is None:
-                # TODO: Raise for this situation if underlay is ISIS.
-                return None
+            if self.shared_utils.underlay_isis is True and isis_system_id_prefix is None:
+                msg = f"'isis_system_id_prefix' is not set on '{self.shared_utils.hostname}' and it is required to set ISIS NET address using node ID"
```

Comment:
```suggestion
                msg = f"'isis_system_id_prefix' is required when 'isis_system_id_prefix' is set to 'node_id'. 'isis_system_id_prefix' was not set for '{self.shared_utils.hostname}'"
```

Reply:
Added this in error msg.
 
 msg = f"'isis_system_id_prefix' is required when `isis_system_id_format` is set to 'node_id'. 'isis_system_id_prefix' was not set for '{self.shared_utils.hostname}'"